Tests/ConfigurationTests/ConfigurationManagerTests.swift

```diff
@@ -0,0 +1,207 @@
+        if configFetched {
+            updateConfigDependencies()
+        }
+    }
+
+    func fetchConfigDependencies(isDebug: Bool) async -> Bool {
+        do {
+            try await fetcher.fetch(.privacyConfiguration, isDebug: isDebug)
+            return true
+        } catch {
+            return false
+        }
+    }
+
+    var onDependenciesUpdated: (() -> Void)?
+    func updateConfigDependencies() {
+        dependencyProvider.privacyConfigData = store.loadData(for: .privacyConfiguration)
+        dependencyProvider.privacyConfigEtag = store.loadEtag(for: .privacyConfiguration)
+        onDependenciesUpdated?()
+    }
+
+    override var presentedItemURL: URL? {
+        return store.fileUrl(for: .privacyConfiguration).deletingLastPathComponent()
+    }
+
+    override func presentedSubitemDidAppear(at url: URL) {
+        guard url == store.fileUrl(for: .privacyConfiguration) else { return }
+        updateConfigDependencies()
+    }
+
+    override func presentedSubitemDidChange(at url: URL) {
+        guard url == store.fileUrl(for: .privacyConfiguration) else { return }
+        updateConfigDependencies()
+    }
+}
+
+struct MockDependencyProvider {
+    var privacyConfigEtag: String?
+    var privacyConfigData: Data?
+}
+
+final class ConfigurationManagerTests: XCTestCase {
+
+    // Shared "UserDefaults" to mock app group defaults
+    var sharedDefaults = MockKeyValueStore()
+
+    override func setUp() {
+        APIRequest.Headers.setUserAgent("")
+        Configuration.setURLProvider(MockConfigurationURLProvider())
+        sharedDefaults.clearAll()
+        MockStoreWithStorage.clearTempConfigs()
+    }
+
+    func makeConfigurationFetcher(store: ConfigurationStoring,
+                                  validator: ConfigurationValidating = MockValidator()) -> ConfigurationFetcher {
+        let testConfiguration = URLSessionConfiguration.default
+        testConfiguration.protocolClasses = [MockURLProtocol.self]
+        return ConfigurationFetcher(store: store,
+                                    validator: validator,
+                                    urlSession: URLSession(configuration: testConfiguration))
+    }
+
+    func makeConfigurationManager(name: String? = nil) -> MockConfigurationManager {
+        let configStore = MockStoreWithStorage(etagStorage: sharedDefaults)
+        let manager = MockConfigurationManager(fetcher: makeConfigurationFetcher(store: configStore),
+                                               store: configStore,
+                                               defaults: sharedDefaults)
+        manager.name = name
+        return manager
+    }
+
+    func testWhenConfigIsFetchedAndStoredDependencyIsUpdated() async {
+        let configurationManager = makeConfigurationManager()
+
+        let configData = Data("Privacy Config".utf8)
+        MockURLProtocol.requestHandler = { _ in (HTTPURLResponse.ok, configData) }
```

Comment:
`MockURLProtocol` is static but isn't resetting itself between tests so I don't trust that we won't get inconsistent behaviour when running the entire test suite - would you be able to add a `tearDown` function to it and call it after running tests? Something like this:

```
static func tearDown() {
    lastRequest = nil
    requestHandler = nil
}
```